Userland/Libraries/LibURL/URL.cpp

```diff
@@ -358,26 +358,44 @@ String URL::serialize_origin() const
 {
```

Comment:
This is quite awkwardly factored, this should be part of LibWeb and actually return an HTML::Origin so it can be used inside of LibWeb - but eh, I am sure push will come to shove once origin handling in general is smoothed out

Reply:
Although, I was also could also make sense to move Origin into LibURL or something instead. Its also a bit awkward that we should be keeping the Origin alongside the BlobURLRegistry - so we will need to be serializing over IPC to the singleton process. Cross spec problems! 😅